src/ms_io.py

```diff
@@ -331,18 +359,102 @@ def read_psms(filename: str) -> pd.DataFrame:
 
 
 def _read_psms_mztab(filename: str) -> pd.DataFrame:
-    raise NotImplementedError
+    """
+    Read mzTab spectrum identifications.
+
+    To correctly have the PSM index the "spectra_ref" column of the mzTab file
+    should use the scan number specification.
+
+    Parameters
+    ----------
+    filename : str
+        The mzTab file name.
+
+    Returns
+    -------
+    pd.DataFrame
+        A Pandas DataFrame with as rows the PSMs and as columns "sequence" and
+        "score", indexed by their spectrum reference in the form of
+        {filename}:scan:{scan}.
+    """
+    run_names, skiplines, nrows = {}, 0, 0
+    with open(filename) as f_in:
+        for i, line in enumerate(f_in):
+            if line.startswith('MTD') and 'ms_run' in line:
+                run_id = line[line.find('ms_run['):line.find('-location')]
+                run_name = os.path.splitext(os.path.basename(
+                    line.rsplit('\t', 1)[1]))[0]
+                run_names[run_id] = run_name
+            elif line.startswith('PSH'):
+                skiplines = i
+            elif line.startswith('PSM'):
+                nrows += 1
+
+    psms = (pd.read_csv(filename, sep='\t', header=skiplines, nrows=nrows + 1)
+            .rename(columns={'search_engine_score[1]': 'score'}))
+    run = psms['spectra_ref'].str.slice(stop=psms['spectra_ref'].str.find(':'))
+    scan = psms['spectra_ref'].str.slice(psms['spectra_ref'].str.rfind('='))
+    psms['spectra_ref'] = run.map(run_names) + ':scan:' + scan
+    return (psms[['spectra_ref', 'sequence', 'score']]
+            .drop_duplicates('spectra_ref')
+            .set_index('spectra_ref')
+            .sort_index())
 
 
 def _read_psms_mzidentml(filename: str) -> pd.DataFrame:
+    """
+    Read mzIdentML spectrum identifications.
+
+    Parameters
+    ----------
+    filename : str
+        The mzIdentML file name.
+
+    Returns
+    -------
+    pd.DataFrame
+        A Pandas DataFrame with as rows the PSMs and as columns "sequence" and
+        "score", indexed by their spectrum reference in the form of
+        {filename}:scan:{scan}.
+    """
     raise NotImplementedError
```

Comment:
This is implemented in Pyteomics, too.

Reply:
Indeed, I'm using Pyteomics to read mzIdentML files into dictionaries, as you can see from the following lines: https://github.com/statisticalbiotechnology/representative-spectra-benchmark/blob/master/src/ms_io.py#L481

But I haven't been able to get all the information I need from there, I need a relevant example file to figure it out. That's why it's not fully implemented at the moment.

Reply:
@levitsky and @bittremieux I recommend to wait until we have the new data coming from the ID pipelines and then we change again the code. The decision should we made basic on: 

- reusability of previous libraries
- performance

We can leave an issue open about reuse of libraries for consitency.

Reply:
OK, it's just that there is a `mzid.DataFrame` shortcut that (naively) packs the dicts from the parser into a DataFrame. If there is something that's missing from the parser output, though, it won't be in the DataFrame either. But we can discuss if it can be fixed on Pyteomics end.

I have not followed everything that's been happening with the pipelines, but I get a sense that we may be moving from mzIdentML to idXML though, in which case it isn't going to be relevant :)